Code/DistGeom/DistGeomUtils.cpp

```diff
@@ -293,14 +295,16 @@ void addImproperTorsionTerms(ForceFields::ForceField *ff,
           break;
       }
 
-      auto *contrib = new ForceFields::UFF::InversionContrib(
-          ff, improperAtom[n[0]], improperAtom[n[1]], improperAtom[n[2]],
+      inversionContribs->addContrib(
+          improperAtom[n[0]], improperAtom[n[1]], improperAtom[n[2]],
           improperAtom[n[3]], improperAtom[4],
           static_cast<bool>(improperAtom[5]), forceScalingFactor);
-      ff->contribs().emplace_back(contrib);
       isImproperConstrained[improperAtom[n[1]]] = 1;
     }
   }
+  if (!inversionContribs->empty()) {
+    ff->contribs().emplace_back(inversionContribs.release());
```

Comment:
It's better to use move here:
```suggestion
    ff->contribs().emplace_back(std::move(inversionContribs));
```